lib/stdlib/src/features.erl

```diff
@@ -0,0 +1,308 @@
+%%
+%% %CopyrightBegin%
+%%
+%% Copyright Ericsson AB 2021-2022. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%
+%% %CopyrightEnd%
+%%
+-module(features).
+
+-export([features/0,
+         feature_info/1,
+         enabled_features/0,
+         is_valid_feature/1,
+         reserved_words/0,
+         reserved_words/1,
+         resword_add_feature/2,
+         resword_add_features/2,
+         resword_remove_feature/2,
+         resword_remove_features/2,
+         enable_feature/1,
+         disable_feature/1,
+         format_error/1,
+         format_error/2]).
+
+-export([features_used/1]).
+
+-on_load(init_features/0).
+
+-type type() :: 'extension' | 'backwards_incompatible_change'.
+-type status() :: {'remove_planned'
+                   | 'inclusion_planned'
+                   | 'removed'
+                   | 'included'
+                   | 'experimental', release()}.
+-type release() :: {Major :: non_neg_integer(),
+                    Minor :: non_neg_integer()}.
+-type error() :: {?MODULE, {'invalid_features', [atom()]}}.
+
+-define(VALID_FEATURE(Feature),
+        (case is_valid_feature(Feature) of
+             false ->
+                 error(invalid_feature, [Feature],
+                       [{error_info,
+                         #{module => ?MODULE,
+                           cause => #{1 => "unknown feature"}}}]);
+             true -> ok
+         end)).
+
+%% Currently know features
+-spec features() -> [atom()].
+features() ->
+    [ifn_expr, maybe_expr, ifnot_expr, unless_expr].
+
+is_valid_feature(Ftr) ->
+    lists:member(Ftr, features()).
+
+-spec feature_info(atom()) -> FeatureInfoMap
+              when
+      Description :: string(),
+      FeatureInfoMap ::
+        #{description := Description,
+          type := type(),
+          status := status(),
+          %% List of additional compiler options to be given to
+          %% activate this feature.  Useful when there is a dependency
+          %% to other features.
+          %% FIXME Should this rather be named dependencies?
+          options := list()
```

Comment:
If it is compiler options only maybe it should be called erlc_options or compiler_options?! Or do we think there could be other types of options in which case options name is very addition friendly.  Calling it dependencies makes me think what kind of dependcies?

Reply:
How will this list of options be used? Will the options be automatically used when the feature is enabled or is it informational?